### ISSUE TITLE
boot/setup selection

### ISSUE THREAD

**Issue by stsp** [dosemu2/dosemu2#1255]
Currently I have a dozen of ~/.dosemu.XXX
dirs (XXX represents the DOS installed there),
which I symlink to ~/.dosemu by hands when
I need to start one.
This is not very user-friendly.
And especially with @jschwartzenberg 's install
scripts finally becoming a reality, I really suspect
the single install dir will become a problem.

So perhaps some switching machinery needs
to be thought up. The easiest start can probably
be to ask @jschwartzenberg to use those .XXX
suffixes when installing, and then add some switch
to dosemu to specify the setup dir.
However, the permanent selection may also be
needed, and for that no simple idea comes to
mind.

Basically I am opening that ticket for a brain-storming,
so please write how would you solve that problem.

**Comment by jschwartzenberg** [comment]
The environment variable `dosemu__hdimage` (note: that's 2x an `_`, underscore) allows you to switch all the drives. So if you run:
`env dosemu__hdimage=~/.dosemu.XXX/drives/* dosemu` it boots using the DOS installation below that `~/.dosemu.XXX`.

**Comment by stsp** [comment]
I don't think it works these days like that.
But there is --Fdrive_c if you want to just
switch the boot drive. But its not the point,
as with different setup dir you also get a
separate .dosemurc.

**Comment by andrewbird** [comment]
I've been using `--Fimagedir` in the test suite, but I also need to switch to using a local dosemu.conf with `-f` and `-n`.

**Comment by stsp** [comment]
Yes, so I know there are switches.
Its not the point.
The points are:
1. Allow multiple parallel installation of different DOSes
2. Allow to boot the needed one easily, w/o tons of
switches or undocumented env vars (that do not work now).
3. Allow to select the needed DOS permanently

**Comment by stsp** [comment]
maybe also 4: write TUI for 2 and 3.

**Comment by stsp** [comment]
And 1 definitely requires modifications in
an install scripts, which is why it should
better be discussed now than when they
are finished.

**Comment by stsp** [comment]
I.e. the question was not how you _do_
solve that problem, but rather how you _would_.
Perhaps 3 should not happen as the only
default should be fdpp?

**Comment by jschwartzenberg** [comment]
> I don't think it works these days like that.

Possibly, but it means you broke that very recently :P

**Comment by jschwartzenberg** [comment]
> But its not the point,
> as with different setup dir you also get a
> separate .dosemurc.

I think something similar to Wine's `WINEPREFIX` environment variable would be appreciated. Maybe `DOSEMUPREFIX`.

**Comment by stsp** [comment]
Well, so far all suggestions I've seen
in the form "wine does this or that"
smelled 90th of the prev milenium. :)
Setting the var is not any better than
using a command-line option.
So for the short terms I can do:
1. Implement option to alter "prefix"
2. Make it so that ~/.dosemu.fdpp
is created by default, and symlinked
to ~/.dosemu

OR, we can forget what wine does
(even in the form of an option, which
is better but no cigar) and for example
go for the directory structure like:
~/.dosemu/fdpp
~/.dosemu/ms-dos_7.01
etc etc.
If I would be implementing from scratches,
I would do something like that. But I
think people will hate me if I start yet
another breakage of existing setups.
(but we are still in alpha, arent we?
who said I won't break everything? :)

**Comment by stsp** [comment]
OK, I actually will think up the
backward-compatible directory
layout for multiple DOSes.
It must:
1. well, be backward-compatible, at least
in most cases
2. contain full DOS names directly in
dir names for an easy TUI writing
(fdpp can be an exception from that rule
perhaps)
3. not pollute the home dir, so contained
within ~/.dosemu
4. allow for master .dosemurc and
per-dos .dosemurc at once (not achievable
with any prefix-alike layout; will allow to
easily select the needed DOS permanently)

**Comment by jschwartzenberg** [comment]
Why are you focused on different DOSs? Maybe user wants to have 10 variants of FDPP or FreeDOS with different boot configurations.

**Comment by stsp** [comment]
Because of your installation scripts!
I think user's desire to have multiple
installs (obviously done by hands) is
a separate problem and should indeed
be solved with the prefix-alike setup.
So lets think of both techniques, right.
But the installers are more important,
plus, as Andrew noted, there are already
the switches to relocate most things.

**Comment by jschwartzenberg** [comment]
I just realized that (some of) these DOSs allow parallel installation beside one another on a real environment too. Not sure how the selection went though.

**Comment by stsp** [comment]
On real environment the selection is done
in a boot sector. The last dos you installed
with sys.com, boots. Another one can
only be "switched in" with its own sys.com,
which is not the user-friendly switching as
sys.com has the same name in all doses.
Also you will get clashes with config/autoexec

**Comment by jschwartzenberg** [comment]
>  Also you will get clashes with config/autoexec

Those are prevented by `dconfig`, `fdconfig`, etc. variants. That avoidance is even in place for DOSEMU2 already :)

**Comment by stsp** [comment]
No, it doest work that way.
config.sys is shared between ms and pc
dos, dconfig is shared by novel and dr dos,
etc. This is just to have a few compatible
dos classes and configure them in a similar
way (as they are compatible).
But installing in parallel the doses from
the same compatibility class, should also
be allowed.

**Comment by stsp** [comment]
In short, our example config.sys that
is similar for ms/pc does't mean the user
wants the same config for ms and pc when
installed in parallel.

**Comment by stsp** [comment]
Implemented w/o permanent selection.
